### PR TITLE
Update e2e to accept eddsa signatures

### PR DESCRIPTION
**Pull request by biscuitdey**
# Description

Updates the E2E test to generate and use Eddsa keypair and signature.

## Related Issue

#759

## How Has This Been Tested

Code will be updated in subsequent PRs to handle the tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bri-3/test/sriUseCase.e2e-spec.ts [hunk 1]
```diff
@@ -4,22 +4,29 @@ import { ethers } from 'ethers';
...
+import * as crypto from 'crypto';
 
 jest.setTimeout(120000);
 
 let accessToken: string;
 let app: INestApplication;
 let server: any;
 
-const supplierBPiSubjectPublicKey =
+const supplierBPiSubjectEcdsaPublicKey =
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/test/sriUseCase.e2e-spec.ts:17]: @biscuitdey can we put all the constants for the e2e test in a constants.ts library file please

**Review by Therecanbeonlyone1969 — DISMISSED** [review]
LGTM except the constants. they should be in a library file.

**Comment by ognjenkurtic** [comment]
@biscuitdey Is this PR supposed to make the e2e test green or is it just a step in that direction?

**Comment by biscuitdey** [comment]
> @biscuitdey Is this PR supposed to make the e2e test green or is it just a step in that direction?

@ognjenkurtic It is a step. Will get fixed in the upcoming PRs.

File: examples/bri-3/test/sriUseCase.e2e-spec.ts [hunk 2]
```diff
@@ -35,6 +42,32 @@ describe('SRI use-case end-to-end test', () => {
     app = moduleFixture.createNestApplication();
     await app.init();
     server = app.getHttpServer();
+
+    const supplierWallet = new ethers.Wallet(
+      supplierBPiSubjectEcdsaPrivateKey,
+      undefined,
```
Comment (skosito) [inline examples/bri-3/test/sriUseCase.e2e-spec.ts:48]: small thing but in js you dont need to provide undefined explicitly as last arg

File: examples/bri-3/test/sriUseCase.e2e-spec.ts [hunk 3]
```diff
@@ -334,3 +376,42 @@ async function fetchBpiAccount(bpiAccountId: string): Promise<any> {
 
   return JSON.parse(getBpiAccountResponse.text);
 }
+
+async function createEddsaPrivateKey(
```
Comment (skosito) [inline examples/bri-3/test/sriUseCase.e2e-spec.ts:380]: these 3 methods looks like are not e2e tests specific, do we use something similar in the rest of codebase, so we can move them in separate module and not duplicate?

File: examples/bri-3/test/sriUseCase.e2e-spec.ts [hunk 4]
```diff
@@ -174,7 +216,7 @@ async function loginAsInternalBpiSubjectAndReturnAnAccessToken(): Promise<string
 
 async function createExternalBpiSubjectAndReturnId(
   bpiSubjectName: string,
-  pk: string,
+  pk: object[],
```
Comment (skosito) [inline examples/bri-3/test/sriUseCase.e2e-spec.ts:219]: can object be more specific here?

File: examples/bri-3/test/sriUseCase.e2e-spec.ts [hunk 5]
```diff
@@ -4,22 +4,29 @@ import { ethers } from 'ethers';
...
+const supplierBPiSubjectEcdsaPrivateKey =
   '0x93b7ed4405c73a1dbd8936e67419ee4e711ed44225aeabe9a5acf49a9ec90e68';
-const buyerBPiSubjectPublicKey =
+const buyerBPiSubjectEcdsaPublicKey =
   '0x04203db7d27bab8d711acc52479efcfa9d7846e4e176d82389689f95cf06a51818b0b9ab1c2c8d72f1a32e236e6296c91c922a0dc3d0cb9afc269834fc5646b980';
-const buyerBPiSubjectPrivateKey =
+const buyerBPiSubjectEcdsaPrivateKey =
   '0x32c8d8f4e53cd1920d1ad22b9d51a7b28216337f2b664fb8d33bbcfc3c455c62';
 
+let supplierBPiSubjectEddsaPublicKey: string;
```
Comment (skosito) [inline examples/bri-3/test/sriUseCase.e2e-spec.ts:26]: `BPi` doesnt work well with my`OCd` :) maybe let's use `Bpi`

**Review by skosito — COMMENTED** [review]
looks good, couple of small comments

File: examples/bri-3/test/sriUseCase.e2e-spec.ts (hunk 4 above)
Comment (biscuitdey) [reply examples/bri-3/test/sriUseCase.e2e-spec.ts:219]: Fixed.

File: examples/bri-3/test/sriUseCase.e2e-spec.ts (hunk 1 above)
Comment (biscuitdey) [reply examples/bri-3/test/sriUseCase.e2e-spec.ts:17]: Done.

File: examples/bri-3/test/sriUseCase.e2e-spec.ts (hunk 3 above)
Comment (biscuitdey) [reply examples/bri-3/test/sriUseCase.e2e-spec.ts:380]: Although we don't have something similar, I added them to test utils, instead of the e2e tests.

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM

**Review by ognjenkurtic — APPROVED** [review]

**Pull request merged**